### PR TITLE
Filescan/8 set dataelement scan status

### PR DESCRIPTION
**Pull request by b-brodie-digdir**
New endpoint for setting FileScanStatus on data elements. 

## Description
See issue for details

## Related Issue(s)
https://github.com/Altinn/altinn-file-scan/issues/8

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manually test contentHash predicate filter
- [x] Relevant automated test added
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Storage/Altinn.Platform.Storage.csproj [hunk 1]
```diff
@@ -9,7 +9,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Altinn.Platform.Storage.Interface" Version="3.11.0" />
+    <PackageReference Include="Altinn.Common.AccessToken" Version="1.2.0" />
+	<PackageReference Include="Altinn.Common.AccessTokenClient" Version="1.1.3" />
```
Comment (SandGrainOne) [inline src/Storage/Altinn.Platform.Storage.csproj:13]: Is Storage in need of the client library?

File: src/Storage/Controllers/DataController.cs [hunk 2]
```diff
@@ -411,6 +415,29 @@ public async Task<ActionResult<DataElement>> OverwriteData(int instanceOwnerPart
...
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Produces("application/json")]
+        public async Task<ActionResult<DataElement>> SetFileScanStatus(
+            Guid instanceGuid,
+            Guid dataGuid,
+            [FromBody] FileScanStatus fileScanStatus)
+        {
+            var success = await _dataRepository.SetFileScanStatus(instanceGuid.ToString(), dataGuid.ToString(), fileScanStatus);
+
+            return success ? Ok() : BadRequest("An error occurred while updating file scan status.");
```
Comment (SandGrainOne) [inline src/Storage/Controllers/DataController.cs:438]: BadRequest should be used when there is something identifiably wrong with the request. The message indicate an unknown error.

File: src/Storage/Altinn.Platform.Storage.csproj (hunk 1 above)
Comment (acn-sbuad) [reply src/Storage/Altinn.Platform.Storage.csproj:13]: Nope, shouldn't be.

File: test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj [hunk 3]
```diff
@@ -161,4 +162,16 @@
     </None>
   </ItemGroup>
 
+	<ItemGroup>
+		<None Update="JWTValidationCert.cer">
```
Comment (acn-sbuad) [inline test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj:166]: Do you need this validation certificate or are the platform-org.*

File: test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj [hunk 4]
```diff
@@ -42,6 +42,7 @@
   <PropertyGroup>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <NoWarn>$(NoWarn);1591</NoWarn>
+    <UserSecretsId>719eaaa3-a93f-43e9-bc80-86e01719b323</UserSecretsId>
```
Comment (acn-sbuad) [inline test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj:45]: is this unique to your personal set up or can it be reused by others working with this project? :) Asking cause I don't know.

File: test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj (hunk 3 above)
Comment (acn-sbuad) [reply test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj:166]: Seems it's not a part of the commit, so only the property that needs removing from this file :)

File: test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj (hunk 4 above)
Comment (SandGrainOne) [reply test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj:45]: The id is the same for everyone.

File: test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj (hunk 3 above)
Comment (b-brodie-digdir) [reply test/UnitTest/Altinn.Platform.Storage.UnitTest.csproj]: Manual project file changes, always fun. Fixed now.

File: src/Storage/Controllers/DataController.cs (hunk 2 above)
Comment (b-brodie-digdir) [reply src/Storage/Controllers/DataController.cs]: Will use error 500 for now until we know how cosmos will handle predicate filter mismatch.

File: src/Storage/Altinn.Platform.Storage.csproj (hunk 1 above)
Comment (b-brodie-digdir) [reply src/Storage/Altinn.Platform.Storage.csproj]: Removed.

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**